proposals/NNNN-change-registration-manager.md

```diff
@@ -0,0 +1,63 @@
+# change-registration-manager
+
+* Proposal: [SDL-0052](NNNN-change-registration-manager.md)
+* Author: [Kujtim Shala](https://www.github.com/kshala)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS]
+
+## Introduction
+
+ChangeRegistration RPC allows an app to manipulate parameters that were send. Including the app name, TTS name and voice synonyms of the app. This proposal aims to make it more convenient for the app developer and to provide an API of a similar style unlike the RPC request.
+
+This proposal comes together with the proposal of localization. It's not required to have both but they would work well together.
+
+## Motivation
+
+Currently the app provides (or can provide) the list of languages it supports. This is done by using the lifecycle configuration. Any app parameter that has to be localized (TTS name, VR names) is localized to the apps default language. The app still has to check manually if it should change the registration and eventually modify the localizable app parameters using the ready handler.
+
+Automatically sending ChangeRegistration inside the SDK is not sufficient. The SDK doesn't know the names for all the other languages.
+
+## Proposed solution
+
+A convenient way for the developer would be to provide an optional method in SDLManagerDelegate. The app can implement this method and modify the lifecycle configuration. If the app developer returns true the manager can reuse the app parameters from the lifecycle and send a ChangeRegistration request. Otherwise if returning false the manager does not execute a registration change.
+
+The manager can decide to call the delegate method by checking the language of the newly connected head unit if it matches one supported language.
+
+The folloing method should be added to the `SDLManagerDelegate`.
+
+ ```objc
+ @protocol SDLManagerDelegate
+ ...
+ @optional
+ - (BOOL)manager:(SDLManager *)manager willChangeRegistrationToLanguage:(SDLLanguage *)language configuration:(SDLLifecycleConfiguration *)configuration; //proposed method
+ @end
+ ```
+
+Optionally the delegate could include a method `manager:didChangeRegistration:` (to follow API design of other protocols in Cocoa Touch).
```

Comment:
Your proposed solution should not have options, it should argue for the proposed solution. That should go under alternatives, or be removed.